### PR TITLE
added epub2txt2

### PR DESCRIPTION
**Pull request by aeblyve**
###### Description of changes

Added epub2txt2. [epub2txt2](https://github.com/kevinboone/epub2txt2) is a simple freestanding C program to extract text from epub files. 

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/text/epub2txt2/default.nix [hunk 1]
```diff
@@ -0,0 +1,23 @@
+{ lib, stdenv, fetchFromGitHub }:
+
+stdenv.mkDerivation rec {
+  pname = "epub2txt2";
+  version = "2.06";
+
+  src = fetchFromGitHub {
+    owner = "kevinboone";
+    repo = pname;
+    rev = "6fd1a332a69b5c1757e3f4d6c27b86920e68a2f8";
```
Comment (NickCao) [inline pkgs/tools/text/epub2txt2/default.nix:10]:
```suggestion
    rev = "refs/tags/v${version}";
```

**Review by NickCao — COMMENTED** [review]
Also please correctly split the changes into two commits:
1. `maintainers: add leonid`
2. `epub2txt2: init 2.06`

**Comment by leonidbelyaev** [comment]
I've made the changes requested.

**Review by NickCao — COMMENTED** [review]
Two more nitpicks:
The first commit should be `maintainers: add leonid`, not `add leonid`
And the build is failing on darwin due to the incorrect use of gcc on darwin, you may fix it by setting `makeFlags = [ "CC:=$(CC)" ];`

**Review by NickCao — APPROVED** [review]
diff LGTM, just waiting for ofborg to finish the build

**Pull request merged**